distributed/scheduler.py

```diff
@@ -502,17 +508,28 @@ def add_worker(self, stream=None, address=None, keys=(), ncores=None,
             if self.ncores[address] > len(self.processing[address]):
                 self.idle.add(address)
 
-            for key in list(self.unrunnable):
-                r = self.restrictions.get(key, [])
-                if address in r or host in r or name in r:
-                    self.transitions({key: 'released'})
```

Comment:
Yeah, I can't think of a cleaner way.  In that case I suppose I'm just bringing it up to bring it up.  Shallowness is a design goal of sorts when convenient.